### PR TITLE
fix: changed index.d.ts imports to reference typescript definitions instead of ts files

### PR DESCRIPTION
**Pull request by mattijsf**
Fixes https://github.com/rnmapbox/maps/issues/2528

### CODE REVIEW

**Comment by mfazekas** [comment]
@mattijsf thanks much. Does our the example app still works without the `lib` folder? For development experience on rnmapbox/maps not for users of the library.

**Comment by mattijsf** [comment]
Ah that is a good one. I guess it might not without running bob build first...

**Comment by mattijsf** [comment]
This requires more work I'm afraid. Good to have these checks in place!

**Comment by mattijsf** [comment]
This change causes the type checking to happen on the actual generated definition files instead of the inferred javascript types which is generating some extra errors. I'm trying to fix those but it not so easy in some cases.

For example. Here and there we tend to use `[number, number]` to indicate x,y coordinates. However we also directly assign `selectedFeature.geometry.coordinates` to one of these props. Which in geojson is modelled as `number[]`. These (and others) now cause some conflicts.

File: example/src/examples/Annotations/ShowPointAnnotation.tsx [hunk 1]
```diff
@@ -43,35 +43,23 @@ const AnnotationWithRemoteImage = ({
...
-  const pointAnnotation = useRef<{ refresh: () => void }>(null);
+  const pointAnnotation = useRef<PointAnnotation>(null);
 
   return (
     <PointAnnotation
       id={id}
       coordinate={coordinate}
       title={title}
       draggable
-      onDrag={(feature: Feature<Point, { id: string }>) =>
```
Comment (mattijsf) [inline example/src/examples/Annotations/ShowPointAnnotation.tsx:54]: The `id` currently not sent by `PointAnnotationManager` to I removed it from the typescript definitions.

File: javascript/components/MarkerView.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ type Props = ViewProps & {
   /**
    * The center point (specified as a map coordinate) of the marker.
    */
-  coordinate: [number, number];
+  coordinate: number[];
```
Comment (mattijsf) [inline javascript/components/MarkerView.tsx:23]: I like the tuple typing in theory but in practice libs like geojson offer points as `number[]` and these can't be assigned to `[number, number]` so it ends up being more of a burden.

**Comment by mfazekas** [comment]
> For me it worked fine after the above changes within the example project because the `prepare` script already generates the `lib` folder.

Yep but switching git branches, etc. doesn't rerun the `prepare` script AFAIK. We could start a watch process, but it means the developer experience became bit more difficult.

**Comment by mattijsf** [comment]
Ah yes that could be causing some annoyances. I'm not sure on how to solve that problem properly except for migrating all javascript files to typescript more quickly but that is not feasible either.

Ideas?

**Comment by mfazekas** [comment]
> Ah yes that could be causing some annoyances. I'm not sure on how to solve that problem properly except for migrating all javascript files to typescript more quickly but that is not feasible either.
> 
> Ideas?

With #2663, we should be really good with js => ts conversion.

Then what remains are those:
- [ ] RasterDemSource
- [ ] RasterSource.js
- [ ] ImageSource
- [ ] UserLocation.js
- [ ] Callout.js
- [ ] Annotation.js
- [ ] offline/*.js
- [ ] snapshot/*.js

so we're close to removing index.d.ts

**Comment by mattijsf** [comment]
Looks like good progress! I think `javascript/classes/*` are missing from that list.

Actually, I started migrating some files some time ago here: https://github.com/mattijsf/rnmapbox-maps/tree/typescript-migration
Including the ones from `javascript/classes/*` but I reverted those in the end due to some difficulties typing the animations correctly.

I did do `geoUtils`, `offline/*.js`, `snapshot/*.js` so feel free to check / cherry-pick those.

**Comment by mattijsf** [comment]
I think we can close this PR in favor of the gradual migration of JS > TS and no longer needing index.d.ts. Which should ultimately fix https://github.com/rnmapbox/maps/issues/2528

**Comment by mfazekas** [comment]
> I think we can close this PR in favor of the gradual migration of JS > TS and no longer needing index.d.ts. Which should ultimately fix https://github.com/rnmapbox/maps/issues/2528

I've merged #2671 so we should be good.

**Pull request closed without merging**